src/lang/term.ml

```diff
@@ -406,11 +418,21 @@ let rec bound_vars_pat = function
 
 let rec free_vars tm =
   match tm.term with
-    | Ground _ | Encoder _ -> Vars.empty
+    | Ground _ -> Vars.empty
     | Var x -> Vars.singleton x
     | Tuple l ->
         List.fold_left (fun v a -> Vars.union v (free_vars a)) Vars.empty l
     | Null -> Vars.empty
+    | Encoder e ->
+        let rec enc (_, p) =
+          List.fold_left
+            (fun v (_, t) ->
+              match t with
+                | `Term t -> Vars.union v (free_vars t)
+                | `Encoder e -> Vars.union v (enc e))
```

Comment:
So, type of `%ogg` is encoder and type of `%vorbis` too and encoders can hold encoders. Gotcha!

Reply:
Yes!